src/base/memory_monitor_sr.cpp

```diff
@@ -358,25 +446,82 @@ namespace cubperf
+    int module_idx = MMON_GET_MODULE_IDX_FROM_STAT_ID (stat_id);
+    int comp_map_idx = MMON_GET_COMP_MAP_IDX_FROM_STAT_ID (stat_id);
+    int idx_map_val, comp_idx, subcomp_idx;
+    bool allocation_at_init = !LOG_ISRESTARTED ();
+
+    idx_map_val = m_module[module_idx]->get_comp_idx_map_val (comp_map_idx);
+    comp_idx = get_comp_idx_from_comp_idx_map (idx_map_val);
+    subcomp_idx = get_subcomp_idx_from_comp_idx_map (idx_map_val);
+
+    m_total_mem_usage += size;
+
+    m_module[module_idx]->add_stat (size, comp_idx, subcomp_idx, allocation_at_init);
+
+    if (thread_p)
+      {
+	tdes = LOG_FIND_CURRENT_TDES (thread_p);
+	tdes->cur_mem_usage += size;
+      }
   }
 
-  int memory_monitor::sub_stat (THREAD_ENTRY *thread_p, MMON_STAT_ID stat_id, uint64_t size)
+  void memory_monitor::sub_stat (THREAD_ENTRY *thread_p, MMON_STAT_ID stat_id, uint64_t size)
   {
-    return 0;
+    LOG_TDES *tdes;
+    TRANID trid = NULL_TRANID;
+    int module_idx = MMON_GET_MODULE_IDX_FROM_STAT_ID (stat_id);
+    int comp_map_idx = MMON_GET_COMP_MAP_IDX_FROM_STAT_ID (stat_id);
+    int idx_map_val, comp_idx, subcomp_idx;
+    bool deallocation_at_init = !LOG_ISRESTARTED ();
+
+    idx_map_val = m_module[module_idx]->get_comp_idx_map_val (comp_map_idx);
+    comp_idx = get_comp_idx_from_comp_idx_map (idx_map_val);
+    subcomp_idx = get_subcomp_idx_from_comp_idx_map (idx_map_val);
+
+    m_total_mem_usage -= size;
+
+    m_module[module_idx]->sub_stat (size, comp_idx, subcomp_idx, deallocation_at_init);
+
+    if (thread_p)
+      {
+	tdes = LOG_FIND_CURRENT_TDES (thread_p);
+	tdes->cur_mem_usage -= size;
+      }
   }
 
-  int memory_monitor::resize_stat (THREAD_ENTRY *thread_p, MMON_STAT_ID stat_id, uint64_t old_size,
-				   uint64_t new_size)
+  void memory_monitor::add_expand_count (MMON_STAT_ID stat_id)
   {
-    return 0;
+    int module_idx = MMON_GET_MODULE_IDX_FROM_STAT_ID (stat_id);
+    int comp_map_idx = MMON_GET_COMP_MAP_IDX_FROM_STAT_ID (stat_id);
+    int idx_map_val, comp_idx, subcomp_idx;
+
+    idx_map_val = m_module[module_idx]->get_comp_idx_map_val (comp_map_idx);
+    comp_idx = get_comp_idx_from_comp_idx_map (idx_map_val);
+    subcomp_idx = get_subcomp_idx_from_comp_idx_map (idx_map_val);
+
+    m_module[module_idx]->add_expand_count (comp_idx);
   }
 
-  int memory_monitor::move_stat (THREAD_ENTRY *thread_p, MMON_STAT_ID src, MMON_STAT_ID dest, uint64_t size)
+  void memory_monitor::resize_stat (THREAD_ENTRY *thread_p, MMON_STAT_ID stat_id, uint64_t old_size,
+				    uint64_t new_size)
   {
-    return 0;
+    this->sub_stat (thread_p, stat_id, old_size);
+    this->add_stat (thread_p, stat_id, new_size);
+    if (new_size - old_size > 0) /* expand */
+      {
+	this->add_expand_count (stat_id);
+      }
+  }
+
+  void memory_monitor::move_stat (THREAD_ENTRY *thread_p, MMON_STAT_ID src, MMON_STAT_ID dest, uint64_t size)
+  {
+    this->sub_stat (thread_p, src, size);
+    this->add_stat (thread_p, dest, size);
```

Comment:
Nit. to all others too.
```suggestion
    sub_stat (thread_p, src, size);
    add_stat (thread_p, dest, size);
```

Reply:
Done.